operator/identity_gc.go

```diff
@@ -50,7 +54,20 @@ func startKvstoreIdentityGC() {
 					metrics.IdentityGCSize.WithLabelValues("deleted").Set(float64(gcStats.Deleted))
 				}
 			}
-			<-gcTimer.After(operatorOption.Config.IdentityGCInterval)
+
+			sleep := operatorOption.Config.IdentityGCInterval
+			if operatorOption.Config.IdentityGCInterval <= gcDuration {
+				log.WithFields(logrus.Fields{
+					logfields.Interval: operatorOption.Config.IdentityGCInterval,
+					logfields.Duration: gcDuration,
+					logfields.Hint:     "Is there a ratelimit configured on the kvstore client or server?",
+				}).Warning("Identity garbage collection took longer than the GC interval")
+				sleep = 0 // Don't sleep because we have a lot of work to do.
+			} else {
+				sleep -= gcDuration
+			}
+			<-gcTimer.After(sleep)
```

Comment:
Nit, we could move this to line 67 and we wouldn't need to run `gcTimer.After` with `0`.

Reply:
Fixed, thanks